roles/mirror_ocp_release/tasks/unpack.yml

```diff
@@ -67,14 +67,41 @@
     # required only for nightly builds
     # using community.general.archive generates the tarball but
     # throws an exception: load_file_common_arguments() unexpected keyword argument 'path'
-    - name: Create a client tarball for el8 on 4.16+ # noqa command-instead-of-module
-      ansible.builtin.shell: >
-        tar czf
-        "{{ mor_cache_dir }}/{{ mor_version }}/openshift-client-linux-amd64-rhel8-{{ mor_version }}.tar.gz"
-        -C "{{ mor_cache_dir }}/{{ mor_version }}"
-        oc
-        kubectl
-      changed_when: true
+    - block:
+        - name: Create a client tarball for el8 on 4.16+ # noqa command-instead-of-module
+          ansible.builtin.shell:
+            cmd: |
+              set -e
+              (
+                flock -x -n 200
+                echo "Lock acquired"
+
+                tar czf \
+                "{{ mor_cache_dir }}/{{ mor_version }}/openshift-client-linux-amd64-rhel8-{{ mor_version }}.tar.gz" \
+                -C "{{ mor_cache_dir }}/{{ mor_version }}" \
+                oc \
+                kubectl
+              ) 200>{{ mor_cache_dir }}/{{ mor_version }}/tar.lock
+
+              rm -f {{ mor_cache_dir }}/{{ mor_version }}/tar.lock
```

Comment:
The removal of the lock file `rm -f {{ mor_cache_dir }}/{{ mor_version }}/tar.lock` after the tarball creation is unnecessary within the block because flock automatically releases the lock when the script exits.

Reply:
Yup. I just remove it so we don't keep "garbage" files.

Do you think it's a better idea just to leave the file there?

Reply:
I believe it's automatically removed by flock

Reply:
I run different tests but in all of them I see the lock is kept.
Also, I'm using it as a mean to have the second job blocked, and waiting for the first one to complete.
The second option would be to have a script looping and trying to get the lock again, but this way I can do it using a builtin module (file) instead of a script.